pennylane/numpy/tensor.py

```diff
@@ -109,6 +109,18 @@ def __repr__(self):
         string = super().__repr__()
         return string[:-1] + ", requires_grad={})".format(self.requires_grad)
 
+    def __array_wrap__(self, out_arr, context=None):
+        out_arr = tensor(out_arr, requires_grad=self.requires_grad)
+        return super().__array_wrap__(out_arr, context)
+
+    def __getitem__(self, *args, **kwargs):
+        item = super().__getitem__(*args, **kwargs)
+
+        if not isinstance(item, tensor):
+            item = tensor(item, requires_grad=self.requires_grad)
+
+        return item
+

```

Comment:
Just to understand, why are these needed now? or is this just for convenience?

Reply:
This is required so that requires_grad info is not lost:
```python
>>> np.array([0.1, 0.2], requires_grad=True)[0]
np.array(0.1, requires_grad=True)
```

This is actually something we should have in core anyway, @co9olguy pointed this out while writing a demo recently.

Reply:
Ah great, I see!

Reply:
It also now matches TF and Torch:

![image](https://user-images.githubusercontent.com/2959003/94147984-a38b5600-fea8-11ea-8131-16d66c7380d2.png)